crates/libtiny_tui/src/key_map.rs

```diff
@@ -0,0 +1,295 @@
+use std::collections::HashMap;
+use std::fmt::Display;
+
+use serde::de::{MapAccess, Unexpected, Visitor};
+use serde::{Deserialize, Deserializer};
+use term_input::{Arrow, Key};
+
+#[derive(Debug, PartialEq)]
+pub(crate) struct KeyMap(HashMap<Key, KeyAction>);
+
+#[derive(Debug, Copy, Clone, Deserialize, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub(crate) enum KeyAction {
+    Disable,
```

Comment:
I think this is not used.

Reply:
It will only be used if someone uses it in their custom config. I added it in case someone wants to disable a keybinding. I can remove if you want

Reply:
Ah, OK, that makes sense. Here's a new suggestion then. "disable" is not really a key action, it indicates lack of a key action, so it seems a bit confusing to have it as a variant in key action type. Since we store the mappings in a `HashMap`, why not just skip the entry for a key/key combo if the value is "disable"? I think it could be easily implemented in `Deserialize for KeyMap`.

Reply:
That's probably possible, but it would definitely over complicate the deserialization since it will have to deserialize to `String`, check if `"disabled"`, then convert to `KeyAction`

Reply:
This kind of thing is easily implemented using GADTs in languages like Haskell and OCaml, but no such luck in Rust.. Anyways, that's fine. I'll merge this later today.